docs/advanced_documentation/native-data-interface.md

```diff
@@ -62,13 +62,45 @@ node['id'] = [1, 2]
 node['u_rated'] = [150e3, 10e3]
 ```
 
+## Columnar data format
+
+We can also represent the contents mentioned `NodeInput` struct in [Structured Array](#structured-array) for only specific attributes.
+This is especially useful when the component in question is for example a transformer which has many default attributes in which case the user can save on a significant amount of memory. Hence we can term it into `NodeInputUrated` which is of `double` type. 
+(Note again, its representation in C++ is not similar to that of `NodeInputUrated`). 
```

Comment:
not sure if we should mention `NodeInputURated` as a type as such, though. maybe just plain old `double` is better

```suggestion
(Note again, its representation in C++ is not similar to that of `NodeInputURated`). 
```